lobsterpy/structuregraph/__init__.py

```diff
@@ -0,0 +1,154 @@
+        self.path_to_charge = os.path.join(self.path_to_calcdir, "CHARGE.lobster")
+        self.path_to_cohpcar = os.path.join(self.path_to_calcdir, "COHPCAR.lobster")
+        self.path_to_icohplist = os.path.join(self.path_to_calcdir, "ICOHPLIST.lobster")
+        self.path_to_madelung = os.path.join(
+            self.path_to_calcdir, "MadelungEnergies.lobster"
+        )
+        self.sg = self.get_decorated_sg()
+
+    def get_decorated_sg(self):
+        """
+        Method to generate graph object decorated with bonding data from lobsterpy
+        Returns:
+            structure graph object
+        """
+        if self.add_additional_data_sg:
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=0,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                filename_blist_sg1=self.path_to_icobilist,
+                id_blist_sg1="ICOBI",
+                filename_blist_sg2=self.path_to_icooplist,
+                id_blist_sg2="ICOOP",
+            )
+
+        else:
+
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=0,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+            )
+
+        chemenvlobster.get_info_cohps_to_neighbors(path_to_COHPCAR=self.path_to_cohpcar)
+
+        (
+            _summed_icohps,
+            list_icohps,
+            _number_bonds,
+            labels,
+            _atoms,
+            _final_isites,
+        ) = chemenvlobster.get_info_icohps_to_neighbors()
+
+        decorated_structure = Charge(self.path_to_charge).get_structure_with_charges(
+            self.path_to_poscar
+        )
+
+        lobster_env = chemenvlobster.get_bonded_structure(
+            structure=decorated_structure, decorate=True, edge_properties=True
+        )
+
+        analyze = Analysis(
+            path_to_charge=self.path_to_charge,
+            path_to_cohpcar=self.path_to_cohpcar,
+            path_to_poscar=self.path_to_poscar,
+            path_to_icohplist=self.path_to_icohplist,
+            path_to_madelung=self.path_to_madelung,
+            whichbonds="all",
+        )
+
+        cba = analyze.condensed_bonding_analysis
+
+        for k, v in cba["sites"].items():
+            for k2, v2 in lobster_env.graph.nodes.data():
+                if v["ion"] == v2["specie"]:
+                    v2["properties"].update({"env": v["env"]})
+
+        for edge_prop in lobster_env.graph.edges.data():
+            for i, icohp in enumerate(list_icohps):
```

Comment:
Are you sure this will work all the time? Numerical problems could show up do to this statement. I think we have to think about a safer way. You could use the identity of the nodes and the image vector to retrieve the correct bond? Or you could at least also check bond length, ICOHP, and identity of the species.

Reply:
I am not sure I understood the comment correctly. maybe we can meet shortly and talk about it.